dashboard/app/models/lesson.rb

```diff
@@ -243,6 +243,19 @@ def render_property(property_name)
     return result
   end
 
+  # If there is a script_level, build_script_level_path will provide the correct url,
+  # even if it's a lockable lesson. Otherwise, we give the url to the student resources
+  # page, and the lesson plan pdf as a backup.
+  def start_url
```

Comment:
Should we add some tests for this method?

Reply:
Done! I've added a couple in lesson_test.rb, one for standard lessons and one for lockable lessons